src/main/java/no/jskdata/Hoydedata.java

```diff
@@ -0,0 +1,111 @@
+package no.jskdata;
+
+import no.jskdata.Downloader;
+import no.jskdata.data.geonorge.File;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.*;
+
+import org.jsoup.Connection;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+/**
+ * Java wrapper around https://hoydedata.no/Laserinnsyn
+ */
+public class Hoydedata extends Downloader {
+
+    private String utmzone = "33"; // default fallback UTM Zone (EUREF-89)
+    private final Set<String> datasetIds = new HashSet<>();
+    private final String baseUrl = "https://hoydedata.no/LaserInnsyn/";
+
+    public Hoydedata() {
+    }
+
+    public void setUtmzone(String utmzone) {
+        this.utmzone = utmzone;
+    }
+
+    public void dataset(String datasetId) {
+        datasetIds.add(datasetId);
+    }
+    public void clear() {
+        datasetIds.clear();
+    }
+
+    private HttpURLConnection openConnectionWithoutCookies(String url) throws IOException {
+        return (HttpURLConnection) new URL(url).openConnection();
+    }
+
+    @Override
+    public void download(Receiver receiver) throws IOException {
+        for (String datasetId : datasetIds) {
+            if (receiver.shouldStop()) {
+                break;
+            }
+            List<File> datasetFiles = getFilesforDataset(datasetId,utmzone);
+            if (datasetFiles.isEmpty()) {
+                break;
+            }
+            for (File file:datasetFiles) {
+                HttpURLConnection conn = openConnectionWithoutCookies(file.downloadUrl);
+                if (conn.getResponseCode() == 200) {
+                    receiver.receive(file.name, conn.getInputStream());
```

Comment:
Would it be possible to use the filename from the `Content-Disposition` http header?

Reply:
And as a consequence, perhaps getFilesforDataset, should return a urls instead of files.